File: src/main/java/com/seaweed/identifier/login/LoginSessionMap.java
package com.seaweed.identifier.login;

import jakarta.servlet.http.HttpSession;

import java.util.ArrayList;
import java.util.HashMap;

public class LoginSessionMap {
    private HashMap<Long, ArrayList<HttpSession>> loginSessionMap = new HashMap();

    public void add(long key,HttpSession session){
        if(loginSessionMap.containsKey(key)) {
            loginSessionMap.get(key).add(session);
        } else {
            var list = new ArrayList<HttpSession>();
            list.add(session);
            loginSessionMap.put(key,list);
        }
    }

    public void remove(long key, HttpSession session){
        if(loginSessionMap.containsKey(key)) {
            loginSessionMap.get(key).remove(session);
        }
    }

}

File: src/main/kotlin/com/seaweed/identifier/auth/LoginSessionMap.kt
package com.seaweed.identifier.auth

import jakarta.servlet.http.HttpSession

class LoginSessionMap {
    private val loginSessionMap : HashMap<Long, ArrayList<HttpSession>> = HashMap();

    fun add(key:Long,session: HttpSession){
        if(loginSessionMap.containsKey(key)) {
            loginSessionMap.get(key)?.add(session);
        } else {
            var list = ArrayList<HttpSession>()
            list.add(session);
            loginSessionMap.put(key,list)
        }
    }

    fun remove(key : Long, session : HttpSession){
        if(loginSessionMap.containsKey(key)) {
            loginSessionMap.get(key)?.remove(session)
        }
    }

}